src/emulate.c

```diff
@@ -1122,24 +1132,8 @@ void rv_step(void *arg)
 }
 
 #if RV32_HAS(SYSTEM)
-static void trap_handler(riscv_t *rv)
-{
-    rv_insn_t *ir = mpool_alloc(rv->block_ir_mp);
-    assert(ir);
-
-    /* set to false by sret/mret implementation */
-    uint32_t insn;
-    while (rv->is_trapped && !rv_has_halted(rv)) {
-        insn = rv->io.mem_ifetch(rv->PC);
-        assert(insn);
-
-        rv_decode(ir, insn);
-        ir->impl = dispatch_table[ir->opcode];
-        rv->compressed = is_compressed(insn);
-        ir->impl(rv, ir, rv->csr_cycle, rv->PC);
-    }
-}
-#endif
+#include "system.c"
```

Comment:
Why did you include this file instead of making it as a new compilation unit?

Reply:
> Why did you include this file instead of making it as a new compilation unit?

Previously, the implementation are compiled in the same unit of `src/emulate.c`. So, simply separate into new file. 

However, a new compilation unit for `system.[ch]` looks good.

Reply:
When attempting to separate into system.[ch], it appears to be non-trivial because `MMU_FAULT_CHECK` depends on the `rv_trap_xx_pgfault` handler, which is a static function within `src/emulate.c`. Also, `trap_handler` depends on `dispatch_table` declared as static within `src/emulate.c`.

The original `#include system.c` is trivial. Or modify `TRAP_HANDLER_IMPL` and `dispatch_table` to non-static?